Log more information on requests that fail at global router level

gsoldevila opened the pull request:
There are certain occasions where Kibana UI shows a blank page (e.g. on this [failed test](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/2999)).
Blank pages can happen when browsers are not able to fetch one or more of the bootstrap resources, e.g. `/bootstrap.js`.
Sometimes, Kibana server-side logs don't contain any information at all as to what caused the failure.

This PR adds a bit more information in the logs, hoping it will help us understanding why some of these requests fail sometimes.

<!--ONMERGE {"backportTargets":["8.10"]} ONMERGE-->

elasticmachine commented:
Pinging @elastic/kibana-core (Team:Core)

jloleysens commented on packages/core/http/core-http-router-server-internal/src/router.ts:
```diff
@@ -200,25 +200,30 @@ export class Router<Context extends RequestHandlerContextBase = RequestHandlerCo
     try {
       kibanaRequest = CoreKibanaRequest.from(request, routeSchemas);
     } catch (e) {
+      this.log.error(`400 Bad Request - ${request.path}${queryString(request.query)}`);
+      this.log.error(e);
```
Minor thing: is it possible to combine these into one log?

With our logger interface we have a `meta` argument that can be passed in:       `this.log.error(e, { url: { query, path } });`. Does that result in anything useful?

gsoldevila commented:
I removed the `queryString` from the message. Let's not log anything not already logged by the http.response.server logger debug mode, and reduce PII concerns.

I also  added the error object to the meta. This way, QA folks can change to JSON logger and have extra info (e.g. trace) if so they wish.

Thus, the PR now simply focusses on NOT having a silent 400 Bad Request, and logging the error message along with the request path in that scenario.

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.10|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/165587">](https://github.com/elastic/kibana/pull/165587)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->
